Fix footer alignment with grid classes and add `width` macro option (#2462)
alphagov/govuk-frontend: CHANGELOG.md

This change was introduced in [pull request #22463: Move template styles from `core` to `objects` layer](https://github.com/alphagov/govuk-frontend/pull/2463).

#### Check your footer displays as expected

> We’ve made some fixes to the alignment of columns within the footer component, so they align with our grid. We have also removed the logic which assumes a two-section layout displays as two-thirds / one-third. Footer sections now display as full-width by default.

Reviewer: ```suggestion
We’ve made some fixes to the alignment of columns within the footer component, so they align with our grid. We've also removed the logic that assumes a 2-section layout displays as a 'two-thirds and one-third' layout. Footer sections now display as full-width by default.
```
Reasons for suggestion:

- style guide say to write out whole numbers unless they're one/1
- style guide says to avoid dashes when they're standing in for a word
- in our Layout guidance, we've referred to 'two-thirds and one third layout'*

*There should be a hyphen between 'one' and 'third' in the guidance - I'll fix that

Author: Thanks @EoinShaughnessy ! Quick question, I'm not sure how changing from `two-section` to `2-section` fits in with "style guide say to write out whole numbers unless they're one/1"? Or am I mis-understanding what that means?

Reviewer: No, totally my bad - sorry! What I should have said was, one is usually the only number we write as a word - all others we can write as numbers. Gah, sorry for the brain fog.

Author: Thanks for clarifying!

> If you are using the Nunjucks macros, check your footer displays as expected and use the `width` macro option to set the desired width of each section.

Reviewer: ```suggestion
If you're using the Nunjucks macros, check your footer displays as expected and use the `width` macro option to set the width you want for each section.
```

> If you are not using the Nunjucks macros, check your footer displays as expected and use the standard design system grid classes on the `govuk-footer__section` element to set the width, for example:

Reviewer: ```suggestion
If you're not using the Nunjucks macros, check your footer displays as expected and use the standard Design System grid classes on the `govuk-footer__section` element to set the width. For example:
```
